lib/request.js

```diff
@@ -121,6 +122,10 @@ function Request (options) {
     simulate: options.simulate || {}
   }
 
+  if (options.customRequestType) {
+    util.inherits(this.constructor, options.customRequestType)
```

Comment:
This alters the constructor, which is a singleton property. This means that after the first execution, all the constructors will have this set. Moreover concurrent usage will be broken too.

Could we move this block next to the other util.inherits?

Reply:
@mcollina you're right about the constructor. I've updated the code so that custom requests are dealt independently from `Request`.

---

About moving this `util.inherits` call down next to the other one: that'd remove the proposed generality, ie users wouldn't be able to provide specific classes/functions from which the request would inherit. Or am I missing something?

In any case this is more of a curiosity than anything. With the new approach this shouldn't be needed.